### DISCUSSION TITLE
Conditional Formatting with Color Scale

### DISCUSSION THREAD

**Discussion by mentor2k1** [PHPOffice/PhpSpreadsheet#2190] [Q&A] [answered]
Hi,

didn't found any example to that, but is it possible to add conditional formatting with a color scale to a cell-range/column, like in the screenshot?
![Bildschirmfoto 2021-06-26 um 11 11 18](https://user-images.githubusercontent.com/1419917/123508408-8a9a1600-d66f-11eb-8ea2-0dc6ce778fa9.png)

Thanks

Oliver

**Comment by mentor2k1** [comment]
Settings of the rule (lowest/highest value):

![Bildschirmfoto 2021-06-26 um 11 34 44](https://user-images.githubusercontent.com/1419917/123508906-8cb1a400-d672-11eb-8834-0fb8abcd8634.png)

**Comment by MarkBaker** [comment]
We currently only support basic Conditional Formatting, and DataBar of the Conditional Formatting Extension Rules; Extension Rules for Gradient and IconSet still need implementing

**Reply by mentor2k1** [reply]
Ok, thanks @MarkBaker. Is there any roadmap, when it will be available?

**Reply by duongp131994** [reply]
Hi @MarkBaker,
I'm building a function to read excel files using PhpSpreadsheet lib, but I don't know how to read the style of cells with conditional formatting applied. can you help me?

$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($file);
$sheet = $spreadsheet->getActiveSheet();
$maxCell = $sheet->getHighestRowAndColumn();
$maxColIndex = \PhpOffice\PhpSpreadsheet\Cell\Coordinate::columnIndexFromString($maxCell['column']);
for ($ci = 1; $ci <= $maxColIndex; $ci++) {
    for ($ri = 1; $ri <= $maxCell['row']; $ri++) {
       $column = \PhpOffice\PhpSpreadsheet\Cell\Coordinate::stringFromColumnIndex($ci);
         //background that i need
       $background = $sheet->getStyle($column . ($ri))->getFill()->getEndColor()->getRGB();
    }
}

$background not by conditional formatting.

**Comment by oleibman** [comment, marked as answer]
Color Scale was added with PR #3738 which was part of release 2.0.0.
